include/engine/routing_algorithms/routing_base.hpp

```diff
@@ -484,6 +484,69 @@ template <class DataFacadeT, class Derived> class BasicRoutingInterface
         unpacked_path.emplace_back(t);
     }
 
+    /**
+     * A duplicate of the above `UnpackEdge` function, but returning full EdgeData
+     * objects for the unpacked path. Used in the tile plugin to find outgoing
+     * edges from a given turn.
+     */
+
```

Comment:
This looks like a shortcoming in FindSmallestEdge in that case. I would say that we have to add that behaviour (template parameter?) to FindSmallestEdge. In that progress, we could also use FindSmallestEdge in UnpackPath over the duplicated code that is used there at the moment.

Technically it should not make a difference, though, as forward/backward only are relevant if two equally valid paths exist.

Consider:

```
a -------------- b -------------- c
|                                 |
---------------- d ----------------
```
It could be that for a route `a` to `c`, both the path using `b` and the path using `d` have the same length. One might only be part of the backward search space, the other one of the forward search space.
In that case the path that is found depends on the order the queues are traversed, but the result is equally valid.
Since the search-graphs only contain traversable edges, both paths are valid.
If you are looking to find both possibilities in a specific way, we would need the distinction using `.forward`